etl/steps/data/meadow/wb/2022/wb_gender.py

```diff
@@ -32,172 +32,175 @@
+        return df, metadata
+
+
+def _load_data(folder_path: Path) -> pd.DataFrame:
+    """Load data from file, as DataFrame."""
+    dtypes = {
+        "Country Name": "category",
+        "Country Code": "category",
+        "Indicator Name": "category",
+        "Indicator Code": "category",
+    }
+    df = pd.read_csv(folder_path / DATA_FILENAME, dtype=dtypes)
+    # Sanity check
+    _sanity_check_columns_data(df)
+    return df
+
+
+def _load_metadata(folder_path: Path) -> pd.DataFrame:
+    """Load metadata from file, as DataFrame."""
+    dtypes = {
+        "Indicator Name": "category",
+    }
+    df = pd.read_csv(folder_path / METADATA_FILENAME, dtype=dtypes)
+    return df
+
+
+def _sanity_check_columns_data(df: pd.DataFrame) -> None:
+    """Run minimal sanity checks on data."""
+    log.info("Checking columns in data are as expected...")
+    assert not df.columns.difference(COLUMNS_KNOWN).any()
+    assert not set(COLUMNS_KNOWN).difference(df.columns)
+
+
+def format_data(df: pd.DataFrame, metadata: pd.DataFrame) -> Any:
+    """Format data.
+
+    Output should be in long format, with four columns:
+        - location: Country name.
+        - variable: Variable name.
+        - year: Year (starting from 1960 til 2021).
+        - value: Value for the variable in the given location and year.
+
+    Also runs checks on the data to see that it is consistent with the metadata.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Data dataframe
+    df : pd.DataFrame
+        Metadata dataframe.
+    """
+    # Get only necessary columns
+    log.info("Reshaping dataframe: Preserving only relevant columns in data...")
+    df = df[COLUMNS_RELEVANT]
+    # Reshape years
+    log.info("Reshaping dataframe: Years in rows...")
+    df = df.melt(["Country Name", "Indicator Name", "Indicator Code"], var_name="year").astype({"year": "category"})
+    # Dropna
+    log.info("Reshaping dataframe: Dropping NaNs...")
+    df = df.dropna(subset="value").reset_index(drop=True)
+    # Column rename
+    log.info("Reshaping dataframe: Renaming columns...")
+    df = df.rename(columns=COLUMNS_DATA_RENAME)
+    # Clean variable name
+    log.info("Reshaping dataframe: Cleaning variable names...")
+    df = df.assign(variable=df["variable"].apply(_clean_variable_name))
+    # Check consistency between data and metadata (primarily review variable names)
+    check_consistency_data_and_metadata(df, metadata)
+    # Final formatting
+    df = final_formatting(df, metadata)
+    return df
+
+
+def format_metadata(df: pd.DataFrame) -> pd.DataFrame:
+    """Format metadata."""
+    # Drop columns
+    log.info("Reshaping dataframe: Preserving only relevant columns in metadata...")
+    df = df.drop(columns=df.filter(regex="Unnamed").columns)
+    # Rename columns
+    log.info("Reshaping dataframe: Renaming columns in metadata...")
+    df.columns = [m.replace(" ", "_").lower() for m in df.columns]
```

Comment:
Just in case you don't know, there is a `catalog.utils.underscore` function to do this. There is also a function `catalog.utils.underscore_table` but (annoyingly) it only works on tables (we should create one that works on dataframes).

Reply:
thanks